tests/attr/test_perm_feature_importance.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+
+import random
+
+import torch
+from captum.attr._core.perm_feature_importance import (
+    PermutationFeatureImportance,
+    permute_feature,
+)
+
+from .helpers.utils import BaseTest, assertTensorAlmostEqual
+
+
+class Test(BaseTest):
+    def test_perm_fn(self):
+        def check_features_are_permuted(inp, perm_inp, mask):
+            unpermuted_features = mask.bitwise_not()
+            self.assertTrue((inp[:, mask] != perm_inp[:, mask]).any())
+            self.assertTrue(
+                (inp[:, unpermuted_features] == perm_inp[:, unpermuted_features]).all()
+            )
+
+        sizes_to_test = [(10,), (4, 5), (3, 4, 5), (6, 7, 8, 9)]
+        for batch_size in [2, 10]:
+            for inp_size in sizes_to_test:
+                inp = torch.randn((batch_size,) + inp_size)
```

Comment:
The nested for -loops might lead to test timeouts. It is better to avoid nested for-loops in the tests and instead split it up into more tests.